tests/conftest.py

```diff
@@ -26,10 +26,15 @@ def remove_data_folder():
     request.addfinalizer(remove_data_folder)
 
 
-@pytest.mark.requires_netcdf
 @pytest.fixture(scope="session")
-def samplecat():
+def samplecat(request):
     """Generate a sample catalog with the tutorial netCDFs."""
+    mark_skip = request.config.getoption("-m")
+    if "not requires_netcdf" == mark_skip or not SAMPLES_DIR.exists():
```

Comment:
```suggestion
    if "not requires_netcdf" in mark_skip or not SAMPLES_DIR.exists():
```
So we cover the cases where other mark parameters are given too ?